autonomx/GeneratorMeta.cpp

```diff
@@ -0,0 +1,162 @@
+#include <QDebug>
+
+#include <QDir>
+#include <QJsonDocument>
+#include <QJsonArray>
+#include <QJsonObject>
+#include <QGuiApplication>
+
+#include "GeneratorMeta.h"
+
+GeneratorMeta::GeneratorMeta(QString type)
+{
+    this->type = type;
+    registerMeta();
+}
+
+QString GeneratorMeta::getName() const
+{
+    return name;
+}
+
+QString GeneratorMeta::getType() const
+{
+    return type;
+}
+
+QString GeneratorMeta::getDescription() const
+{
+    return description;
+}
+
+QVariantMap GeneratorMeta::getFieldTree() const
+{
+    return fieldTree;
+}
+
+QVariantMap GeneratorMeta::getHelpRacks() const
+{
+    return helpRacks;
+}
+
+QVariantMap GeneratorMeta::getEnumLabels() const
+{
+    return enumLabels;
+}
+
+void GeneratorMeta::registerMeta() {
+    // the previous way of doing it,
+    // via a static reference to the generators folder
+//    #ifdef Q_OS_MAC
+//        QString basePath = QCoreApplication::applicationDirPath() + "/generators/" + this->type;
+//    #else
+//        QString basePath = QDir::currentPath() + "/generators/" + this->type;
+//    #endif
+
+    // the clutch way of doing it,
+    // by importing assets in the .qrc and reading them from there
```

Comment:
Using the qrc resource system is the best way to make sure our application is cross-platform. If we don't do this, we'll need to implement the installation of the JSON file in a different on each platform, and that might take multiple days to figure it out, if not weeks. We would also need to set up multiple virtual machines, ideally.